src/coord/src/coord.rs

```diff
@@ -4374,16 +4465,25 @@ impl Coordinator {
 
     /// Finalizes a list of dataflows and then broadcasts it to all workers.
     async fn ship_dataflows(&mut self, dataflows: Vec<DataflowDesc>) {
```

Comment:
Are there cases where this would be called with multiple distinct instance identifiers? It feels like the previous moments where we may have created "some dataflows" are probably all in the same instance, unless surprising things? I ask mostly because the contained logic seems plenty complicated allowing for this generality.

Reply:
The generality seemed simple enough to me, but certainly glad to restrict it to only supporting a single instance.